awx/main/access.py

```diff
@@ -934,7 +957,7 @@ def can_run_ad_hoc_commands(self, obj):
 
     def can_attach(self, obj, sub_obj, relationship, *args, **kwargs):
         if relationship == "instance_groups":
-            if self.user.can_access(type(sub_obj), "read", sub_obj) and self.user in obj.organization.admin_role:
+            if self.user in sub_obj.use_role:
```

Comment:
Inventory - you're attaching an IG to the inventory. In that case you should need admin permission to the inventory, and on first reading here I'm not seeing that.